controllers/kyma_controller.go

```diff
@@ -125,8 +125,9 @@ func (r *KymaReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 func (r *KymaReconciler) reconcile(ctx context.Context, kyma *v1beta2.Kyma) (ctrl.Result, error) {
 	if r.SyncKymaEnabled(kyma) {
 		remoteClient := remote.NewClientWithConfig(r.Client, r.KcpRestConfig)
-		if err := remote.InitializeSyncContext(&ctx, kyma,
-			r.RemoteSyncNamespace, remoteClient, r.RemoteClientCache); err != nil {
+		ctxWithSync, err := remote.InitializeSyncContext(ctx, kyma,
```

Comment:
I'm thinking maybe it's better to relocate this `InitializeSyncContext` (renaming if needed) as a common one before passing into `reconcile(ctx context.Context, ...)`. And inside the function body, depends on if `SyncKymaEnabled`, return back context with syncContext or without, so that in this step, we don't need to deal with context anymore.

Reply:
Refactored to external method and added docs on special cases on the method.